server/src/internalClusterTest/java/org/elasticsearch/reservedstate/service/FileSettingsServiceIT.java

```diff
@@ -84,7 +86,10 @@ private void writeJSONFile(String node, String json) throws Exception {
         FileSettingsService fileSettingsService = internalCluster().getInstance(FileSettingsService.class, node);
 
         Files.createDirectories(fileSettingsService.operatorSettingsDir());
-        Files.write(fileSettingsService.operatorSettingsFile(), Strings.format(json, version).getBytes(StandardCharsets.UTF_8));
+        Path tempFilePath = fileSettingsService.operatorSettingsFile().resolveSibling("settings.tmp");
```

Comment:
Maybe write to a true temp file, using createTempFile(), so that it is not in the same directory that is being watched?